extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -121,23 +124,24 @@ export class AmpAutocomplete extends AMP.BaseElement {
     this.maxEntries_ = this.element.hasAttribute('max-entries') ?
       parseInt(this.element.getAttribute('max-entries'), 10) : null;
 
-    return this.mutateElement(() => {
+    return Promise.all([dataPromise, this.mutateElement(() => {
```

Comment:
`buildCallback` already runs in a mutate context ([source](https://github.com/ampproject/amphtml/blob/master/contributing/building-an-amp-extension.md#buildcallback)), so you shouldn't need that additional `mutateElement` :)
Also, that same document recommends loading the data from the `layoutCallback` instead

Reply:
Nice, TIL about vsync mutate context in the `buildCallback`. I didn't know that doc had that information in it either

Reply:
Nice, TIL about vsync mutate context in the `buildCallback`. I didn't know that doc had that information in it either